src/Hng.Web/Controllers/ProductController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Hng.Application.Features.Products.Commands;
+using Hng.Application.Features.Products.Dtos;
+using Hng.Application.Features.Products.Responses;
+using Hng.Application.Features.UserManagement.Dtos;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
+    [ApiController]
+    [Route("api/v1/products")]
+    public class ProductController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+
+        public ProductController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(typeof(ProductDto), StatusCodes.Status201Created)]
+        [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status500InternalServerError)]
+
+        public async Task<ActionResult<UserDto>> CreateProduct([FromBody] ProductCreationDto body)
+        {
+            try
+            {
+                var command = new CreateProductCommand(body);
+                var response = await _mediator.Send(command);
+
+                return CreatedAtAction(nameof(CreateProduct), new
+                {
+                    status = "success",
+                    message = "Product created successfully",
+                    data = response
+                });
+            }
+            catch (ValidationException ex)
+            {
```

Comment:
What validation are you doing?
Take this out. Enforce validation using Annotations in `ProductCreationDto`